### DISCUSSION TITLE
Allow users to create / manage subscriptions in the CP

### DISCUSSION THREAD

**Discussion by leevigraham** [craftcms/commerce#2487] [Plugin ideas]
### Description

It would be great if users could subscribe / manage existing subscriptions in the CP. My first thought is a new "subscriptions" tab in the user profile section.

In my use case I'm allowing users to login to the CP to create their own user generated content. They manage their account from the CP, not front end templates which is typically the case.

**Comment by lukeholder** [comment]
Decided that this lends itself to a plugin. You can add a tab to the user profile page like we do for the customer tab. See the following example:

https://github.com/craftcms/commerce/blob/develop/src/Plugin.php#L729-L730

https://github.com/craftcms/commerce/blob/develop/src/services/Customers.php#L816-L854

You can then take a look at the example front-end templates as a guide on adding the subscription management features the tab you add.

**Comment by leevigraham** [comment]
@lukeholder has this be reopened because it's being reconsidered for core?

**Comment by nfourtythree** [comment]
Hi @leevigraham 

As per @lukeholder comment, we have decided that this is better suited as a plugin due to all the implications that come with it.

It is essentially building a front-end subscription process in the CP. We have no plans to add this feature to Commerce at this moment in time.

It is being left open for visibility/until there is a plugin that solves this case.

Thanks.

**Comment by comtyler** [comment]
Hi @nfourtythree

Could you provide some thoughts on the implications of implementing the subscription creation process in the control panel? Initial thoughts are you would need to choose:

-  a user (customer)
- the subscription plan
- a payment source (or the ability to create a new one)

What are some of the edge cases here?

I also imagine the interfacing could be similar to the "New Order" creation process in the control panel.

Thanks!

**Comment by leevigraham** [comment]
@comtyler I ended up creating a new custom tab in the user admin and hooking it up to Stripe subscriptions (completely bypassing Craft Commerce)

![image](https://user-images.githubusercontent.com/25124/207159756-642d3ea5-8b44-4852-a7ad-b99ed852d90d.png)

![image](https://user-images.githubusercontent.com/25124/207159906-69d9cd86-c497-41ee-8690-7b0ad3e7d839.png)
